.github/workflows/build.yml

```diff
@@ -0,0 +1,73 @@
+name: Build
+
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+
+env:
+    REGISTRY: ghcr.io
+
+jobs:
+
+    set-env:
+        runs-on: [ubuntu-22.04]
+        outputs:
+            CI_IMAGE : ${{ steps.common.outputs.CI_IMAGE }}
+        steps:
+            - name: Export common variables.
+              id  : common
+              run : |
+                  echo "CI_IMAGE=${{ env.REGISTRY }}/${{ github.repository }}/kokkos-utils:latest" >> $GITHUB_OUTPUT
+
+    build-image:
+        needs: [set-env]
+        runs-on: [ubuntu-22.04]
+        container:
+            image: docker:latest
+        permissions:
+            packages: write
+        steps:
+            - name: Checkout code.
+              uses: actions/checkout@v4
+
+            - name: Log in to Docker registry.
+              run : |
+                  docker login -u ${{ github.actor}} -p ${{ secrets.GITHUB_TOKEN }} ${{ env.REGISTRY }}
```

Comment:
Reading 

- https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions

I saw that there are also "docker/login-action" and "docker/build-push-action" actions

- https://github.com/docker/login-action
- https://github.com/docker/build-push-action

It seems this could allow us to do these things within an `ubuntu-latest` image.

Something like:

```
name: Login to GitHub Container Registry
        uses: docker/login-action@v3
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

It seems the build-push-action works with buildx and all the parameters that we need.

Reply:
Done for `login`.

Reply:
I will not do it for the `build` part (yet).